manticore/core/smtlib/expression.py

```diff
@@ -690,21 +682,24 @@ def cast(self, possible_array):
             return arr
         raise ValueError  # cast not implemented
 
-    def cast_index(self, index):
+    def cast_index(self, index: Union[int, "BitVec"]) -> Union["BitVecConstant", "BitVec"]:
         if isinstance(index, int):
             # assert self.index_max is None or index >= 0 and index < self.index_max
             return BitVecConstant(self.index_bits, index)
-        assert isinstance(index, BitVec) and index.size == self.index_bits
+        assert index.size == self.index_bits
         return index
 
-    def cast_value(self, value):
+    def cast_value(
```

Comment:
Hmm, with the old behavior, if you try to pass in something that isn't `int`, `str`, `bytes`, or `BitVec`, you'll get an `AssertionError` on line 705. The new behavior will instead try to cast it to an int, which may or may not succeed. I don't _think_ we ever do anything that weird with this function, so that's probably fine, but I'd like a second opinion. @feliam, any thoughts?

Reply:
I ran our test suite through my type tracer. These are all the types we test `value` as:
`{'BitVecAdd', 'BitVecExtract', 'BitVecVariable', 'int', 'str', 'bytes', 'BitVecConcat', 'BitVecConstant', 'ArraySelect'}`

All of those are covered by the type hint, so I think we're okay